### PR TITLE
refactor: change AuthorityManage and remove old height

### PR DESCRIPTION
**Pull request by rev-chaos**

### CODE REVIEW

File: src/state/process.rs [hunk 1]
```diff
@@ -90,32 +90,32 @@ where
         wal_engine: Arc<W>,
     ) -> (Self, UnboundedReceiver<VerifyResp>) {
         let (tx, rx) = unbounded();
-        let mut auth = AuthorityManage::new();
-        auth.update(&mut authority_list, false);
+        let mut authority = AuthorityManage::new();
+        authority.update(&mut authority_list);
 
         let state = State {
```
Comment (KaoImin) [inline src/state/process.rs]: This looks ugly. Rename authorigy.

Comment (KaoImin) [inline src/state/process.rs]: This looks ugly. Rename authorigy.

Comment (KaoImin) [reply src/state/process.rs]: Rename the variable `authority`, and the struct below will be formatted.

File: src/state/process.rs [hunk 2]
```diff
@@ -449,12 +449,7 @@ where
     /// interval. Since it is possible to have received and cached the current height's proposals,
     /// votes and quorum certificates before, these should be re-checked as goto new height.
     /// Finally, trigger SMR to goto new height.
-    async fn goto_new_height(
-        &mut self,
-        ctx: Context,
-        status: Status,
-        get_last_flag: bool,
```
Comment (yejiayu) [inline src/state/process.rs]: Why remove this flag?

Comment (KaoImin) [reply src/state/process.rs]: Overlord will not handle messages that below self height, so that the authority list of the last height is useless.

**Pull request merged**